spring-cloud-gcp-samples/spring-cloud-gcp-sql-sample/src/test/java/com/example/CloudSqlApplicationTests.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2017-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example;
+
+import com.example.CloudSqlApplicationTests.CloudSqlApplicationTestConfiguration;
+import org.junit.runner.RunWith;
+
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.cloud.gcp.autoconfigure.sql.CloudSqlJdbcInfoProvider;
+import org.springframework.cloud.gcp.autoconfigure.sql.DatabaseType;
+import org.springframework.cloud.gcp.autoconfigure.sql.DefaultCloudSqlJdbcInfoProvider;
+import org.springframework.cloud.gcp.autoconfigure.sql.GcpCloudSqlProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.junit4.SpringRunner;
+
+/**
+ * Runs the {@link SqlApplicationTestCase} on a Cloud SQL database backend.
+ *
+ * @author Daniel Zou
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(
+		webEnvironment = WebEnvironment.RANDOM_PORT,
+		classes = { CloudSqlApplicationTestConfiguration.class, SqlApplication.class },
+		properties = {
+				"spring.cloud.gcp.sql.database-name=code_samples_test_db",
+				"spring.cloud.gcp.sql.instance-connection-name=spring-cloud-gcp-ci:us-central1:testmysql",
+				"spring.datasource.password=test",
+				"spring.datasource.continue-on-error=true",
+				"spring.datasource.initialization-mode=always"
+		})
+public class CloudSqlApplicationTests extends SqlApplicationTestCase {
+
+	@Configuration
+	static class CloudSqlApplicationTestConfiguration {
+		@Bean
+		public CloudSqlJdbcInfoProvider mySqlJdbcInfoProvider(
+				GcpCloudSqlProperties gcpCloudSqlProperties) {
+			CloudSqlJdbcInfoProvider provider = new DefaultCloudSqlJdbcInfoProvider(gcpCloudSqlProperties,
+					DatabaseType.MYSQL);
```

Comment:
What if we made `DatabaseType` an optional property in `CloudSqlJdbcInfoProvider`? I think it would make this code a lot simpler, and it would enable users to more easily force the database type when both dependencies are present.
cc/ @artembilan

Reply:
I think we really can expose a `dataBaseType` as a configuration property in the `GcpCloudSqlProperties`.
And it really should have an effect in the `GcpCloudSqlAutoConfiguration` when we have both dependencies on classpath.

However looking at this from the target application perspective, I don't think that this is going to have some value: you connect to MySql or PsotgreSql in cloud from your application. Nothing more. So, my position to stay here with what we have so far. Or create MySql and PsotgreSql as separate samples. The point of samples to be as simple as possible with one-stop-shop guide to use in the target project.

Reply:
I see your point. I would be open to either keeping it this way (for the sake of reducing duplicate code) or writing a separate sample (for the sake of being simpler). One can argue that this sample is now confusing because the user will have to comment/uncomment the SQL starter from the pom.xml. So I am open to redoing this PR for a separate sample too; let me know what you guys think.